### PR TITLE
elf_find_string.c

### PR DESCRIPTION
**Pull request by vanbako**
The RE C code for elf_find_string

### CODE REVIEW

**Comment by smx-smx** [comment]
I think we should move the reconstructed source files into a separate directory (something like `src_decomp`), then add CMake instructions to build them.
Something like `add_library(xzdecomp ${SOURCES})` or similar

File: dasm.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+#include "xzre.h"
+
+//  7 = 0x80
+//  8 = 0x78
+//  9 = 0x70
+// 10 = 0x68
+// 11 = 0x58
+// 12 = 0x50
+BOOL find_lea_instruction(u8 *code_start, u8 *code_end, u64 displacement, u64 padding4, u64 padding5, u64 padding6, dasm_ctx_t *ctx, u64 padding8, u64 padding9, u64 padding10, );
```
Comment (smx-smx) [inline dasm.c:16]: the prototype is incorrect. it doesn't match with the one in `xzre.h`

File: dasm.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    uint64_t var_58h;
+    int64_t var_50h;
+
+// to check because this is not correct yet    
+    uVar4 = 0;
+    if (secret_data_append_from_call_site(0x7c, 5, 6, 0) != 0) {
+        memset(ctx, 0x00, 0x16);
+        piVar3 = piVar3 + uVar4 * -8 + 4;
+        for (; code_start < code_end; ++code_start) {
+            int iVar1 = x86_dasm(ctx, code_start, code_end);
```
Comment (smx-smx) [inline dasm.c:29]: `ctx` here points to the context on the stack, it's not passed in

File: dasm.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            if (((iVar1 != 0) && (var_58h == 0x10d) && ((var_6fh & 0x7) == 1)) &&
```
Comment (smx-smx) [inline dasm.c:30]: these are variables on the stack too, part of `dasm_ctx_t`

File: elf.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
+/**
+ * @file elf.c
+ * @author Koen Van Bastelaere
+ * @brief Elf functions
+ * 
+ */
+
+#include "xzre.h"
+
+u8 *elf_find_string(elf_info_t *elf_info, EncodedStringId encoded_string_id, u8 *code_start) {
```
Comment (smx-smx) [inline elf.c:10]: this is incorrect. `encoded_string_id` is passed as an in-out pointer (look for the correct prototype in `xzre.h`)

File: elf.c [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    }
+    return NULL;
+}
+
+Elf64_Sym *elf_symbol_get_addr(elf_info_t *elf_info, EncodedStringId encoded_string_id)
+{
+    Elf64_Sym *elf_symbol = elf_symbol_get(elf_info, encoded_string_id, 0);
+    if (elf_symbol != NULL) {
+        if (((elf_symbol + 8) == 0) || ((elf_symbol + 6) == 0)) {
```
Comment (smx-smx) [inline elf.c:42]: these are members of `Elf64_Sym`, and should be named accordingly (offset 6: `st_shndx` and offset 8: `st_value`)

File: elf_find_string.c [hunk 6]
```diff
@@ -0,0 +1,36 @@
+/**
+ * @file elf_find_string.c
+ * @author Koen Van Bastelaere
+ * @brief
+ * 
+ */
+
+#include "xzre.h"
+
+u8 *elf_find_string(elf_info_t *elf_info, EncodedStringId encoded_string_id, u8 *code_start) {
```
Comment (smx-smx) [inline elf_find_string.c:10]: duplicated? it's also in `elf.c`

**Review by smx-smx — CHANGES_REQUESTED** [review]
Thanks for taking the time to submit this.
Unfortunately, the code has some issues and wouldn't work properly as it is right now.
It has to be aligned with `xzre.h` prototypes, first of all.
Fields also need to be renamed to reflect the containing type.

**Comment by vanbako** [comment]
Sorry that second commit shouldn't have been in this PR.
I don't know why VSC did that

**Pull request closed without merging**